evalml/tests/dependency_update_check/make_deps_diff.sh

```diff
@@ -0,0 +1,3 @@
+allow_list=$(cat core-requirements.txt requirements.txt | grep -oE "^[a-zA-Z0-9\-_]+" | paste -d "|" -s -)
```

Comment:
The grep picks up both scikit-learn and scikit-optimize as just "scikit". Is that ok?

![image](https://user-images.githubusercontent.com/41651716/100755510-42a56000-33ba-11eb-806a-fc28efac3370.png)

Reply:
@freddyaboulton yes because we're using `grep -E` which uses a regex, and "scikit" will match both sklearn and skopt.